### DISCUSSION TITLE
Is there an abstraction to have a list of records from a table similar to the Firebase's onSnapshot?

### DISCUSSION THREAD

**Discussion by joaomelo** [supabase/supabase#9675] [Questions] [answered]
Hi there,

Is there an abstraction to have a list of records from a table similar to the Firebase's [onSnapshot ](https://firebase.google.com/docs/firestore/query-data/listen#listen_to_multiple_documents_in_a_collection)?

While using the realtime feature of the Supabase js library, if I understood it correctly, there is no initial load of the selected records upon subscription, and changes do not provide the select result again but expose specific data from the change. 

Am I correct to say that the onSnapshot experience would need to be developed by putting together a first fetch of all records inside an array or hash and then updating the variable based on realtime events?

**Comment by GaryAustin1** [comment, marked as answer]
Yes,  you should start your subscription, then when connection is confirmed load your initial data and use events to keep it updated.   If your subscription ever drops, you need to repeat the process or you could lose any data change while the subscription was reconnecting.

**Reply by joaomelo** [reply]
Thank you, @GaryAustin1,
You are awesome,

Let me just confirm if my understanding of your answer is correct...

To keep the list of records of a table in sync, I need to do the first load and by myself implement updates corresponding to each event, and there is **no** feature in the Supabase client that enables this synchronization, abstracting the dealing with individual events and a first load like the pseudocode below. 

```
const cities = [];
client.from('cities-table').onTableChangeIncludingFirstLoadDoThis((myUpdatedCompleteTableData) => {
  myUpdatedCompleteTableData.forEach((record) => cities.push(record));
});
```

**Reply by GaryAustin1** [reply]
All realtime will do is send changes to a row.  It does not deal with larger data structures or joins.  You need to handle setting up any rows you want to monitor in an object each time you start up the subscription AND then update that structure per event.

This is a bit dated, but shows the idea I used:  https://github.com/supabase/supabase/discussions/5641#discussioncomment-2292166

**Reply by joaomelo** [reply]
Thanks again @GaryAustin1
